smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/model/ShapeExt.kt

```diff
@@ -123,15 +123,17 @@ fun Model.getNestedShapes(serviceShape: ServiceShape): Set<Shape> {
         .select(this)
 }
 
-fun Model.getNestedShapes(memberShape: MemberShape): Set<Shape> {
+fun Model.getNestedErrors(serviceShape: ServiceShape): Set<StructureShape> {
     return Selector
-        .parse("member [id='${memberShape.id}'] ~> *")
+        .parse("service[id=${serviceShape.id}] ~> structure[trait|error]")
         .select(this)
+        .map { it as StructureShape }
+        .toSet()
 }
 
-fun Model.getNestedShapes(shape: Shape): Set<Shape> {
+fun Model.getNestedShapes(memberShape: MemberShape): Set<Shape> {
     return Selector
-        .parse("operation [id='${shape.id}'] ~> *")
+        .parse("member [id='${memberShape.id}'] ~> *")
```

Comment:
The diff here is confusing, but I deleted
`fun Model.getNestedShapes(shape: Shape): Set<Shape>` and replaced it with
`fun Model.getNestedErrors(serviceShape: ServiceShape): Set<StructureShape>`.